README.md

```diff
@@ -1,11 +1,48 @@
 # ALBA
+This is a Rust library of _Approximate Lower Bound Arguments_ proposed in the [paper](https://iohk.io/en/research/library/papers/approximate-lower-bound-arguments/), May 2024, Eurocrypt'24 by _Pyrros Chaidos, Aggelos Kiayias, Leonid Reyzin, Anatoliy Zinovyev_.
 
-Implementation of _Approximate Lower Bound Arguments_ from the [paper](https://iohk.io/en/research/library/papers/approximate-lower-bound-arguments/) published by IOG Research:
+### Introduction
+ALBA enables a prover who has a large collection of data to convince a verifier that their set includes at least a minimum number of elements that meet a specific condition, called a predicate, even though the prover only reveals a subset of the data.
+By approximating a lower bound on the prover's knowledge, ALBA makes use of a controlled gap between the size of prover's actual knowledge and the threshold of the verifier checks they know.
+This design results in highly efficient proofs, achieving nearly optimal proof sizes in both non-interactive and distributed environments.
+ALBA's primary applications include large-scale decentralized signature schemes.
+It is particularly well-suited for decentralized or blockchain scenarios, where it enhances communication efficiency among multiple provers sharing witness.
 
-> **Approximate Lower Bound Arguments**,  _Pyrros Chaidos, Prof Aggelos Kiayias, Leonid Reyzin, Anatoliy Zinovyev_, May 2024, Eurocrypt'24
+In a decentralized voting system, participants (voters) submit votes that support different options or candidates.
+To validate the results without revealing all individual votes, an ALBA protocol could be used.
+Each voter's choice can be considered an _element_ that meets a certain predicate (e.g., a valid vote for a candidate).
+Instead of tallying every vote publicly, ALBA allows an aggregator (like an election authority) to generate a compact proof showing that a sufficient number of valid votes has been cast for each candidate to meet the required threshold for a decision (such as reaching a quorum or winning a majority).
+Combining ALBA with zero-knowledge technology such as _zkSNARKs_, we can efficiently keep individual votes private while enabling quick, efficient validation of the voting outcome, making it particularly useful for secure, private, and scalable voting systems where privacy and efficiency are critical.
+
+The paper introduces various ALBA protocol constructions tailored to different needs.
```

Comment:
This doesn't make sense imo. You say that there are multiple constructions "tailored to different needs", but then you talk about the basic construction and the construction with prehashing. Both are not superior in any way to the (third) bounded DFS construction, so they do not serve any special needs. The decentralized and weighted variants do, however.